### PR TITLE
Code style enforcement

### PR DESCRIPTION
**Pull request by adejanovski**
I added a checkstyle file which should comply with the Cassandra style, with a style check during maven build. All source files were restyled accordingly.
The CCM scenarios are now part of the CI runs on push (not the nightly ones as I saw they could be a little flaky).
Running the CCM tests locally is easier now as it only requires ccm to be installed and doesn't expect a cluster to be running already.

I'll squash the commits before merge.

### CODE REVIEW

File: src/test/resources/ccm/node1/conf/jvm8-server.options [hunk 1]
```diff
@@ -74,3 +74,8 @@
 -XX:GCLogFileSize=10M
 
 # The newline in the end of file is intentional
```
Comment (rzvoncek-ds) [inline src/test/resources/ccm/node1/conf/jvm8-server.options:76]: Is the new line really there?

**Review by rzvoncek-ds — CHANGES_REQUESTED** [review]
I had a look through the files. Aside from the squas-me commits, there's zip files in `src/test/resources/ccm/node*/logs` that I don't think we need to commit. Maybe we should even add a .gitignore entry for them.

File: src/test/resources/ccm/node1/conf/jvm8-server.options (hunk 1 above)
Comment (adejanovski) [reply src/test/resources/ccm/node1/conf/jvm8-server.options:76]: This comment is there in the base files from the Cassandra repo 🤷  https://github.com/apache/cassandra/blob/trunk/conf/jvm8-server.options#L76

I guess it's mentioning the empty line above, and not one that would be below... 😅

**Comment by adejanovski** [comment]
Thanks for the review @rzvoncek-ds!

The log files you're seeing are already removed. They were showing up in the diff because I initially added them.

File: src/test/resources/ccm/node1/conf/jvm8-server.options (hunk 1 above)
Comment (michaelsembwever) [reply src/test/resources/ccm/node1/conf/jvm8-server.options:76]: file ending with a newline character is a thing. 

cassandra.yaml didn't have it for a long time and it was causing pretty of git grief for people.

**Pull request merged**